### PR TITLE
Restart debugging

### PR DESCRIPTION
**Pull request by alhom**

### CODE REVIEW

File: parameters.h [hunk 1]
```diff
@@ -81,6 +81,7 @@ struct Parameters {
    
    static bool writeInitialState;           /*!< If true, initial state is written. This is useful for debugging as the restarts are always written out after propagation of 0.5dt in real space.*/
    static Real saveRestartWalltimeInterval; /*!< Interval in walltime seconds for restart data*/
+   static uint saveRestartDtInterval;       /*!< Interval in dts for restart data*/
```
Comment (markusbattarbee) [inline parameters.h]: interval in dts is what? :)

File: vlasiator.cpp [hunk 2]
```diff
@@ -736,10 +736,11 @@ int main(int argn,char* args[]) {
       // Combined with checking of additional load balancing to have only one collective call.
       phiprof::start("compute-is-restart-written-and-extra-LB");
       if (myRank == MASTER_RANK) {
-         if (  (P::saveRestartWalltimeInterval >= 0.0
+         if (  (P::saveRestartWalltimeInterval >= 0.0 
             && (P::saveRestartWalltimeInterval*wallTimeRestartCounter <=  MPI_Wtime()-initialWtime
-               || P::tstep == P::tstep_max
+               || P::tstep == P::tstep_max 
                || P::t >= P::t_max))
+            || (P::saveRestartDtInterval > 0 && P::tstep % P::saveRestartDtInterval == 0) //NB! short-circuit && guards from fp expection
```
Comment (markusbattarbee) [inline vlasiator.cpp]: Let's clearly use either time or timesteps

**Review by markusbattarbee — CHANGES_REQUESTED** [review]
restart dt saving needs some cleanup

File: vlasovsolver/vlasovmover.cpp [hunk 3]
```diff
@@ -348,7 +348,14 @@ void calculateSpatialTranslation(
    
    // Mapping complete, update moments and maximum dt limits //
 momentCalculation:
-   calculateMoments_R(mpiGrid,localCells,true);
+   if(P::isRestart && P::tstep_min == P::tstep && dt == 0.0)
```
Comment (markusbattarbee) [inline vlasovsolver/vlasovmover.cpp:351]: So I think that instead of this check, we should revert the changes at 
https://github.com/markusbattarbee/vlasiator-public/blob/25efbf93afae067b246aef3550f4bb46c1693860/vlasiator.cpp#L518
So that zero-timestep translation and acceleration are performed only if it's not a restart.

Comment (alhom) [reply vlasovsolver/vlasovmover.cpp:351]: Right, no notification from github for me, still :(. That change had the purpose of preserving the population-specific moment calculations, otherwise the first bulk output after restart is garbled for pop moments. Not very relevant, since the moment recalculation error is small anyhow.

Comment (markusbattarbee) [reply vlasovsolver/vlasovmover.cpp:351]: Ok. That's a bit of a special case where output is triggered right after a restart happened and before a single propagation has taken place - unlikely, but can indeed happen. (And I think what'd happen is that file would have already been written before the restart, but would get overwritten by the new run)

This however is something that could be tackled by re-ordering or adding a momentscalculation call to output if we're at the first timestep instead of adding lots of ifs inside the momentcalculation function. And should be fixed in regular Vlasiator as well! :)

Comment (alhom) [reply vlasovsolver/vlasovmover.cpp:351]: Yes, that seems a more reasonable solution.

**Comment by alhom** [comment]
NB: needs some more reverting.

File: vlasiator.cpp (hunk 2 above)
Comment (alhom) [reply vlasiator.cpp]: saveRestartTimestepInterval now.
